### DISCUSSION TITLE
Dark mode on TeslaMate Dashboard

### DISCUSSION THREAD

**Discussion by spacecosmos** [teslamate-org/teslamate#236] [Ideas]
The title kind of says it all. Any way to implement it? :)

I think it would look good with Grafana already being in **dark mode** as well. (Just a thought and not a very important feature of course.)

**Comment by adriankumpf** [comment]
Yeah, a dark mode would be awesome! Shouldn't be that difficult to implement. I'll see what I can do.

**Comment by tobiasehlert** [comment]
This should maybe be not too hard to implement with the [CSS media feature prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme)

**Comment by Niek** [comment]
Fixed in https://github.com/adriankumpf/teslamate/pull/461

**Comment by roadrash2108** [comment]
PR #461 never got merged, so this issue is still open. Is this issue still worked on?

**Comment by khelge** [comment]
Any news here? Still bleeds my eyes to death looking at the dashboard before I get into Grafana which is dark and nice

**Comment by LightBringer81** [comment]
I'm also for dark mode. I check my stats regularly at night, when the room is dark and my eyes would love to be able to check teslamate in a darker version. Preferably with options to a dark blue and a complete black design.

**Comment by gryghst72** [comment]
Bump

**Comment by NirKli** [comment]
Bump

I made a small POC, not sure if its the best solution for dark mode (controlled by the OS setting).
I'm using invert color, one of the reason why I'm using invert colors is because there's no dark mode support for the map.

@spacecosmos @gryghst72 

(main)
![image](https://user-images.githubusercontent.com/45952578/175993997-76ab5f09-bc46-4ebf-968f-803f63956397.png)


(settings)

![image](https://user-images.githubusercontent.com/45952578/175992791-6197982c-24a4-4a85-b735-d20f96990317.png)

**Reply by tobiasehlert** [reply]
There are plugins för Leaflet JS, with which you can get a dark map.

Checkout this page to see how the map could look like in other themes:
https://leaflet-extras.github.io/leaflet-providers/preview

I think that [Stadia.AlidadeSmoothDark](https://leaflet-extras.github.io/leaflet-providers/preview/#filter=Stadia.AlidadeSmoothDark) would be the best alternative :)

**Reply by NirKli** [reply]
> There are plugins för Leaflet JS, with which you can get a dark map.
> 
> 
> 
> Checkout this page to see how the map could look like in other themes:
> 
> https://leaflet-extras.github.io/leaflet-providers/preview
> 
> 
> 
> I think that [Stadia.AlidadeSmoothDark](https://leaflet-extras.github.io/leaflet-providers/preview/#filter=Stadia.AlidadeSmoothDark) would be the best alternative :)
> 
> 

It looks better than my solution 

![image](https://user-images.githubusercontent.com/45952578/176141796-0e511287-2323-491e-8879-6a3594d77085.jpeg)

**Reply by NirKli** [reply]
> There are plugins för Leaflet JS, with which you can get a dark map.
> 
> Checkout this page to see how the map could look like in other themes: https://leaflet-extras.github.io/leaflet-providers/preview
> 
> I think that [Stadia.AlidadeSmoothDark](https://leaflet-extras.github.io/leaflet-providers/preview/#filter=Stadia.AlidadeSmoothDark) would be the best alternative :)

I think this map theme looks a little empty when using geo-fence, so I replaced it with the old map theme with invert color.
I have attached some videos, I think its not the best solution but its much better in the dark 

Overview page

[overview.webm](https://user-images.githubusercontent.com/45952578/176453793-b18c5513-2df6-4491-9ed1-8383f74bf63f.webm)


[settings.webm](https://user-images.githubusercontent.com/45952578/176455488-e269d1fa-38e9-44be-aab3-a1cdfa831f25.webm)


[geofence.webm](https://user-images.githubusercontent.com/45952578/176455545-f927e0e8-4a94-4086-9d66-35dc95477a0b.webm)

**Reply by tobiasehlert** [reply]
Well, it's not that often that you are inside the geo-fence or settings page anyways, so if it does look slightly empty is not a reason for it.
The map should be using a dark theme in addition to the rest of the gui and the front page will look good enough I think.

I do at least like the image you added in your previous comment https://github.com/adriankumpf/teslamate/discussions/236#discussioncomment-3038376 regarding the theme.. on top of that with some other coloring on the rest of the page it would look good I think :)

**Reply by NirKli** [reply]
> Well, it's not that often that you are inside the geo-fence or settings page anyways, so if it does look slightly empty is not a reason for it. The map should be using a dark theme in addition to the rest of the gui and the front page will look good enough I think.
> 
> I do at least like the image you added in your previous comment [#236 (reply in thread)](https://github.com/adriankumpf/teslamate/discussions/236#discussioncomment-3038376) regarding the theme.. on top of that with some other coloring on the rest of the page it would look good I think :)

[map.webm](https://user-images.githubusercontent.com/45952578/176473950-4ae67902-a744-4c2f-bdf9-a80ff81d29ea.webm)

**Comment by NirKli** [comment]
@tobiasehlert @spacecosmos @Niek 

Created PR :)
[Added dark mode support for teslamate UI #2730](https://github.com/adriankumpf/teslamate/pull/2730)

**Comment by NirKli** [comment]
@tobiasehlert 

Like you said, its feeling more relaxing to the eyes when looking greyish colors

![image](https://user-images.githubusercontent.com/45952578/178155214-0b69b3d5-5ef7-421b-b565-57a7d793fcd6.png)
